app/controllers/publications_controller.rb

```diff
@@ -28,82 +28,71 @@ def index
-
-          if params[:format] =~ /csv/i
-            csv_string = generate_csv_report author
-          else
-            matching_records = author.publications.order('publications.id').page(page).per(per).select('publications.pub_hash')
-          end
+        end
+        unless params[:format] =~ /csv/i
+          matching_records = author.publications.order('publications.id').page(page).per(per).select(:pub_hash)
         end
       end
       logger.debug("Found #{matching_records.length} records")
 
       respond_to do |format|
         format.json do
-          bibjson = wrap_as_bibjson_collection(description, matching_records, page, per)
-          self.response_body = JSON.dump(bibjson)
+          render json: wrap_as_bibjson_collection(description, matching_records, page, per)
         end
         format.csv do
-          render csv: csv_string, filename: 'author_report', chunked: true
+          render csv: generate_csv_report(author), filename: 'author_report', chunked: true
         end
       end
     end
   end
 
-  # desc "Look up existing records by title, and optionally by author, year and source"
+  # Look up existing records by title, and optionally by author, year and source
   def sourcelookup
     all_matching_records = []
+    msg = 'Sourcelookup of '
     if params[:doi]
-      logger.info("Sourcelookup of doi #{params[:doi]}")
-      sources = [Settings.sciencewire_source]
-      all_matching_records += DoiSearch.search params[:doi]
+      msg << "doi #{params[:doi]}"
+      logger.info(msg)
+      all_matching_records += DoiSearch.search(params[:doi])
     elsif params[:pmid]
-      logger.info("Sourcelookup of pmid #{params[:pmid]}")
-      sources = [Settings.sciencewire_source, Settings.pubmed_source]
-      all_matching_records += PubmedHarvester.search_all_sources_by_pmid params[:pmid]
+      msg << "pmid #{params[:pmid]}"
+      logger.info(msg)
+      all_matching_records += PubmedHarvester.search_all_sources_by_pmid(params[:pmid])
     else
       raise ActionController::ParameterMissing, :title unless params[:title].presence
-
-      source = params.fetch(:source, Settings.manual_source + '+' + Settings.sciencewire_source)
-      logger.info("Executing source lookup for title #{params[:title]} with sources #{source}")
-
-      sources = source.split('+')
-
-      if sources.include?(Settings.sciencewire_source)
-        all_matching_records += ScienceWireClient.new.query_sciencewire_for_publication(params[:firstname], params[:lastname], params[:middlename], params[:title], params[:year], params.fetch(:max_rows, 20).to_i)
-        logger.debug(" -- sciencewire (#{all_matching_records.length})")
+      msg << "title '#{params[:title]}'"
+      logger.info(msg)
+      if Settings.WOS.enabled
+        query = "TI=#{params[:title]}"
+        query += " AND PY=#{params[:year]}" if params[:year]
+        wos_matches = WebOfScience.queries.user_query(query).to_a # TODO: limit
```

Comment:
limit at the query level is tricky, right? since the WoS client doesn't support passing this as a parameter?  But if I read the code correctly, it is already defaulting to 100 ?  or will it just keep retrieving in chunks of 100 (this could be a problem if you typed in a really vague title)

Reply:
Right, WoS client doesn't expose this functionality.  I have a separate branch where I started to add that.  The ScienceWire integration default is **20**.  WoS default is ALL.  Given enough hits for common strings (and serial trips back to WoS), the CAP client will HTTP timeout before we make a response.

Reply:
ok, so the plan is to address in this workcycle which seems important to keep those timeouts from happening

Reply:
But not in this PR.